sale_advance_payment/models/sale.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017 Omar Castiñeira, Comunitea Servicios Tecnológicos S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class SaleOrder(models.Model):
+
+    _inherit = "sale.order"
+
+    account_payment_ids = fields.One2many(
+        "account.payment", "sale_id", string="Pay sale advanced", readonly=True
+    )
+    amount_residual = fields.Float(
+        "Residual amount",
+        readonly=True,
+        compute="_compute_advance_payment",
+        store=True,
+    )
+    payment_line_ids = fields.Many2many(
+        "account.move.line",
+        string="Payment move lines",
+        compute="_compute_advance_payment",
+        store=True,
+    )
+    advance_payment_status = fields.Selection(
+        selection=[
+            ("not_paid", "Not Paid"),
+            ("paid", "Paid"),
+            ("partial", "Partially Paid"),
+        ],
+        string="Advance Payment Status",
+        store=True,
+        readonly=True,
+        copy=False,
+        tracking=True,
+        compute="_compute_advance_payment",
+    )
+
+    @api.depends(
```

Comment:
Maybe a good improvement idea here is to split several computed values in separate fields (mls, advance_amount, amount_residual).

That allow a better comprehension of logic, and a better inheritance possibility.

Reply:
@MateuGForgeFlow Could you put a TODO here or a ROADMAP for this ?

Reply:
@rousseldenis Done! Added a ROADMAP. Thanks!